src/averageAges.js

```diff
@@ -55,7 +76,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childF = people.filter(child => onlyWithSon ? people.some(person => (
+    child.mother === person.name
+  ))
+    && child.sex === 'm'
+    : people.some(
+      person => child.mother === person.name
+    ));
+
+  const averageAge = childF.reduce(
+    (sum, child) => sum + (child.born - people.find(mother => (
+      child.mother === mother.name
+    )).born), 0) / childF.length;
```

Comment:
Agree this would be more structured and readable
```
  const averageAge = childF.reduce((sum, child) => {
    const mother = ...;
    const ageDifference = ...;
    
    return sum + ageDifference;
  } , 0) / childF.length;
```